agent/hcp/manager.go

```diff
@@ -108,6 +112,12 @@ func (m *Manager) Run(ctx context.Context) {
 
 	// main loop
 	for {
+		// Check for configured management token from HCP. It MUST NOT override the user-provided initial management token.
+		if hcpManagement := m.cfg.CloudConfig.ManagementToken; len(hcpManagement) > 0 {
```

Comment:
Its probably ok to have this run on every loop iteration for now, but I worry that as we expand the scope of the manager we'll want a dedicated signaling mechanism for this.